telegram/ext/_basepersistence.py

```diff
@@ -73,33 +73,30 @@ class BasePersistence(Generic[UD, CD, BD], ABC):
     * :meth:`get_chat_data`
     * :meth:`update_chat_data`
     * :meth:`refresh_chat_data`
+    * :meth:`drop_chat_data`
     * :meth:`get_user_data`
     * :meth:`update_user_data`
     * :meth:`refresh_user_data`
+    * :meth:`drop_user_data`
     * :meth:`get_callback_data`
     * :meth:`update_callback_data`
     * :meth:`get_conversations`
     * :meth:`update_conversation`
     * :meth:`flush`
 
-    If you don't actually need one of those methods, a simple ``pass`` is enough. For example, if
-    you don't store ``bot_data``, you don't need :meth:`get_bot_data`, :meth:`update_bot_data` or
-    :meth:`refresh_bot_data`.
-
-    Warning:
-        Persistence will try to replace :class:`telegram.Bot` instances by :attr:`REPLACED_BOT` and
-        insert the bot set with :meth:`set_bot` upon loading of the data. This is to ensure that
-        changes to the bot apply to the saved objects, too. If you change the bots token, this may
-        lead to e.g. ``Chat not found`` errors. For the limitations on replacing bots see
-        :meth:`replace_bot` and :meth:`insert_bot`.
+    If you don't actually need one of those methods, a simple :keyword:`pass` is enough.
+    For example, if you don't store ``bot_data``, you don't need :meth:`get_bot_data`,
+    :meth:`update_bot_data` or :meth:`refresh_bot_data`.
 
     Note:
-         :meth:`replace_bot` and :meth:`insert_bot` are used *independently* of the implementation
-         of the ``update/get_*`` methods, i.e. you don't need to worry about it while
-         implementing a custom persistence subclass.
+        You should avoid saving :class:`telegram.Bot` instances, this is because if you change
+        the bots token, this may lead to e.g. ``Chat not found`` errors. To prevent this, the
+        persistence *may* automatically use :attr:`bot` to replace the bot. Otherwise, you must
+        manually set the bot with :meth:`set_bot` upon loading of the data.
```

Comment:
```suggestion
        You should avoid saving :class:`telegram.Bot` instances. This is because if you change
        e.g. the bots token, this won't propagate to the serialized instances and may lead exceptions.
        
        To prevent this, the implemtation may use :attr:`bot` to replace bot instances with a placeholder before serialization and insert :attr:`bot` back when loading the data. Since :attr:`bot` will be set when the process starts, this will be the up-to-date bot instance.
        
        If the persistence implementation does not take care of this, you should make sure not to store any bot instances in the data that will be persisted. E.g. in case of :class:`TelegramObject`, one may call :meth:`set_bot` to ensure that shortcuts like `Message.reply_text` are available.
```